pypesto/visualize/model_fit.py

```diff
@@ -72,62 +72,106 @@ def visualize_optimized_model_fit(
     axes: `matplotlib.axes.Axes` object of the created plot.
     None: In case subplots are saved to file
     """
-    x = result.optimize_result.list[start_index]["x"][
-        pypesto_problem.x_free_indices
-    ]
-    objective_result = pypesto_problem.objective(x, return_dict=True)
-
-    simulation_df = rdatas_to_simulation_df(
-        objective_result[RDATAS],
-        pypesto_problem.objective.amici_model,
-        petab_problem.measurement_df,
-    )
-
-    # handle flattened PEtab problems
-    petab_problem_to_plot = petab_problem
-    if unflattened_petab_problem:
-        simulation_df = petab.core.unflatten_simulation_df(
-            simulation_df=simulation_df,
-            petab_problem=unflattened_petab_problem,
-        )
-        petab_problem_to_plot = unflattened_petab_problem
 
-    # plot
-    axes = plot_problem(
-        petab_problem=petab_problem_to_plot,
-        simulations_df=simulation_df,
-        **kwargs,
-    )
+    amici_models, objective_results = [], []
 
-    non_quantitative_data_types = get_petab_non_quantitative_data_types(
-        petab_problem
-    )
+    # get amici model
+    if hasattr(pypesto_problem.objective, AMICI_MODEL):
+        x = result.optimize_result.list[start_index]["x"][
+            pypesto_problem.x_free_indices
+        ]
 
-    if non_quantitative_data_types is not None:
-        if (
-            ORDINAL in non_quantitative_data_types
-            or CENSORED in non_quantitative_data_types
-        ):
-            axes = plot_categories_from_pypesto_result(
-                result,
-                start_index=start_index,
-                axes=axes,
+        objective_results.append(
+            pypesto_problem.objective(x, return_dict=True)
+        )
+        amici_models.append(pypesto_problem.objective.amici_model)
+    elif hasattr(pypesto_problem.objective, "_objectives"):
+        # iterate over objectives to find the amici models in the AggregatedObjective
+        x = result.optimize_result.list[start_index]["x"]
+
+        for objective in pypesto_problem.objective._objectives:
+            if hasattr(objective, AMICI_MODEL):
```

Comment:
I'd emit a warning for any non-AmiciObjective encountered here. (Or less preferred add to the documentation that non-AmiciObjective are silently ignored.)